### DISCUSSION TITLE
`_FORTIFY_SOURCE` requirements

### DISCUSSION THREAD

**Discussion by dotnwat** [scylladb/seastar#2342] [Q&A] [answered]
Seastar undefines `_FORTIFY_SOURCE` on all it's dependents. However, the comment here in `CMakeLists.txt` suggests it is only needed to be set on the compilation of `thread.cc` (indeed that seems to be the only file containing `longjmp`).

```cmake
# We disable _FORTIFY_SOURCE because it generates false positives with longjmp() (src/core/thread.cc)

target_compile_options (seastar
  PUBLIC
    -U_FORTIFY_SOURCE)
```

We'd like to broaden the scope of `_FORITFY_SOURCE` by considering the removal of this public flag since it becomes viral in a Seastar application. We tried removing it, and things seemed ok, but we are wondering if we may be missing something important / would Seastar be opposed to removing this?

**Comment by avikivity** [comment, marked as answer]
3d75bab9b4870c63cd8ee8561fde067043aab4d7 (2015)

If it no longer complains, we can remove it. Running the test suite in all modes should be sufficient.

**Reply by dotnwat** [reply]
Thanks. Additional discussion occurring in https://github.com/scylladb/seastar/pull/2344.
